.github/workflows/lint.yml

```diff
@@ -1,11 +1,59 @@
 name: Lint
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - '**'
 
 jobs:
-  build:
-    runs-on: ubuntu-20.04
+  lint:
+    name: Python ${{ matrix.python-version }} - ${{ matrix.os }} - ${{ github.event_name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - 3.8
```

Comment:
should we consider doing more versions here? or do you think just 3.8 is sufficient?

Reply:
I restricted the scope of the github actions to linting and formatting only. Since it's not responsible for any tests, I don't see a reason to run on a matrix of versions or operating systems. The linters should run the same on every version, since they're only doing static analysis